File: 2022/day-01/c/day-01-part-1.c
/// \brief AoC 2022 Day 01 Part 1 Solution
///
/// Author: Tyler Swann (oraqlle.net@gmail.com)
///
/// Date: 09/07/2023
///
/// License: Apache-2.0 license
///
/// Copyright (c) 2023 - present
/// \file day-01-part-1.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    char* path = "../day-01-input.txt";
    FILE* file = fopen(path, "r");

    if (file == NULL) {
        puts("Error opening file!");
        exit(1);
    }

    size_t sum = 0;
    size_t max = 0;

    char current[20];

    while (fgets(current, sizeof current, file)) {

        if (strcmp(current, "\n") == 0) {
            if (max < sum)
                max = sum;

            sum = 0;
        } else {
            current[strcspn(&current[0], "\n")] = '\0';
            sum += (size_t)strtoull(&current[0], NULL, 10);
        }
    }

    printf("Result: %zu\n", max);
    fclose(file);

    return 0;
}

File: 2022/day-01/elixir/day-01-part-1.exs
# \brief AoC 2022 Day 01 Part 1 Solution
#
# Author: Tyler Swann (oraqlle.net@gmail.com)
#
# Date: 28/06/2023
#
# License: Apache-2.0 license
#
# Copyright (c) 2023 - present
# \file day-01.exs

defmodule Day01 do
  @path Path.expand(Path.join("..", "day-01-input.txt"))

  def main do
    case File.read(@path) do
      {:error, reason} ->
        IO.puts(reason)
        exit(:error)

      {:ok, data} ->
        max =
          data
          |> String.split("\n\n")
          |> Enum.map(fn chunk ->
            chunk
            |> String.split("\n")
            |> Enum.map(fn str -> if str != "", do: String.to_integer(str), else: 0 end)
            |> Enum.sum()
          end)
          |> Enum.max()

        IO.puts("Result: #{max}")
    end
  end
end

Day01.main()
